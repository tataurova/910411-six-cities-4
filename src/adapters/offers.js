export const getOffer = (data) => {
  return {
    id: data.id,
    title: data.title,
    city: data.city.name,
    cityCoordinates: [data.city.location.latitude, data.city.location.longitude],
    cityZoom: data.city.location.zoom,
    coordinates: [data.location.latitude, data.location.longitude],
    type: data.type,
    price: data.price,
    rating: data.rating,
    premium: data.is_premium,
    favorite: data.is_favorite,
    photo: data.preview_image,
    bedrooms: data.bedrooms,
    adults: data.max_adults,
    additional: data.goods,
    details: data.description,
    owner: {
      photo: data.host.avatar_url,
      name: data.host.name,
      isSuper: data.host.is_pro,
    },
    photos: data.images,
    reviews: [ // todo reviews from server
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  };
};
