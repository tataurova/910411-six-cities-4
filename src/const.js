export const PLACES_COUNT = 312;

export const MapSettings = {
  city: [52.38333, 4.9],
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30],
  },
  zoom: 12,
};
