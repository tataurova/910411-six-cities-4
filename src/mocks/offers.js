export default [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.960361, 6.967974],
    type: `Apartment`,
    price: 120,
    rating: 4,
    premium: true,
    photo: `img/apartment-01.jpg`,
    bedrooms: 3,
    adults: 6,
    additional: [`Wifi`, `Heating`, `Kitchen`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: true,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.913361, 6.9509739999999995],
    type: `Private room`,
    price: 80,
    rating: 4,
    premium: false,
    photo: `img/room.jpg`,
    bedrooms: 2,
    adults: 4,
    additional: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-max.jpg`,
      name: `Max`,
      isSuper: false,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];

/* eslint-disable */
export const serverOffers = [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.960361,
      longitude: 6.967974,
    },
    type: `Apartment`,
    price: 120,
    rating: 4,
    is_premium: true,
    preview_image: `img/apartment-01.jpg`,
    bedrooms: 3,
    max_adults: 6,
    goods: [`Wifi`, `Heating`, `Kitchen`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 1,
      avatar_url: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      is_pro: true,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.913361,
      longitude: 6.9509739999999995,
    },
    type: `Private room`,
    price: 80,
    rating: 4,
    is_premium: false,
    preview_image: `img/room.jpg`,
    bedrooms: 2,
    max_adults: 4,
    goods: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 2,
      avatar_url: `img/avatar-max.jpg`,
      name: `Max`,
      is_pro: false,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];
