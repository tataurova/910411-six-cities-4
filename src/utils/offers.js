export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  return {
    offer: offers.filter((item) => item.id === idNumber)[0],
    nearOffers: offers.filter((item) => item.id !== idNumber),
  };
  // for (let key in offers) {
  //   if (offers.hasOwnProperty(key)) {
  //     const offer = offers[key].filter((item) => item.id === idNumber)[0];
  //     if (offer) {
  //       return {
  //         offer,
  //         nearOffers: offers[key].filter((item) => item.id !== offer.id),
  //       };
  //     }
  //   }
  // }
  // return {};
};
