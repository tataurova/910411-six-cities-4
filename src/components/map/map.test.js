import React from "react";
import renderer from "react-test-renderer";
import Map from "./map.jsx";
import {MapSettings} from "../../const";
import offers from "../../mocks/offers.js";

describe(`<Map />`, () => {
  it(`Should Map render correctly`, () => {
    const cityOffers = offers[`Amsterdam`];
    const tree = renderer
      .create(
          <Map
            offers={cityOffers}
            mapSettings={MapSettings}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
