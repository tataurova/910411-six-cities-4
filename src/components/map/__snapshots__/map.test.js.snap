// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> Should Map render correctly 1`] = `<div />`;
